src/pages/home/report/comment/TextCommentFragment.tsx

```diff
@@ -1,56 +1,48 @@
 import Str from 'expensify-common/lib/str';
-import PropTypes from 'prop-types';
+import {isEmpty} from 'lodash';
 import React, {memo} from 'react';
+import type {StyleProp, TextStyle} from 'react-native';
 import Text from '@components/Text';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
-import withWindowDimensions, {windowDimensionsPropTypes} from '@components/withWindowDimensions';
 import ZeroWidthView from '@components/ZeroWidthView';
+import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
-import compose from '@libs/compose';
+import useWindowDimensions from '@hooks/useWindowDimensions';
 import convertToLTR from '@libs/convertToLTR';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import * as EmojiUtils from '@libs/EmojiUtils';
-import reportActionFragmentPropTypes from '@pages/home/report/reportActionFragmentPropTypes';
-import reportActionSourcePropType from '@pages/home/report/reportActionSourcePropType';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
+import type {OriginalMessageSource} from '@src/types/onyx/OriginalMessage';
+import type {Message} from '@src/types/onyx/ReportAction';
 import RenderCommentHTML from './RenderCommentHTML';
 
-const propTypes = {
+type TextCommentFragmentProps = {
     /** The reportAction's source */
-    source: reportActionSourcePropType.isRequired,
+    source: OriginalMessageSource;
 
     /** The message fragment needing to be displayed */
-    fragment: reportActionFragmentPropTypes.isRequired,
+    fragment: Message;
 
     /** Should this message fragment be styled as deleted? */
-    styleAsDeleted: PropTypes.bool.isRequired,
-
-    /** Text of an IOU report action */
-    iouMessage: PropTypes.string,
+    styleAsDeleted: boolean;
 
     /** Should the comment have the appearance of being grouped with the previous comment? */
-    displayAsGroup: PropTypes.bool.isRequired,
+    displayAsGroup: boolean;
 
     /** Additional styles to add after local styles. */
-    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]).isRequired,
-
-    ...windowDimensionsPropTypes,
+    style: StyleProp<TextStyle>;
 
-    /** localization props */
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    iouMessage: undefined,
+    /** Text of an IOU report action */
+    iouMessage?: string;
 };
 
-function TextCommentFragment(props) {
+function TextCommentFragment({fragment, styleAsDeleted, source, style, displayAsGroup, iouMessage = ''}: TextCommentFragmentProps) {
```

Comment:
Is this default type needed?
```suggestion
function TextCommentFragment({fragment, styleAsDeleted, source, style, displayAsGroup, iouMessage}: TextCommentFragmentProps) {
```